### PR TITLE
Terminate task

### PR DESCRIPTION
**Pull request by boma96**
Added support for terminate task

```
builder.AddTask(
                wf => wf.Terminate,
                wf => new() { TerminationStatus = TerminationStatus.Failed, WorkflowOutput = new { Property = "Test" } }
            );
```

### CODE REVIEW

**Review by ognjenkatic — CHANGES_REQUESTED** [review]
Just one suggestion

File: src/ConductorSharp.Engine/Model/TerminateTaskModel.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        public const string Failed = "FAILED";
+    }
+
+    public class TerminateTaskInput : IRequest<NoOutput>
+    {
+        [JsonProperty("workflowOutput")]
+        public dynamic WorkflowOutput { get; set; }
+
+        [JsonProperty("terminationStatus")]
+        public string TerminationStatus { get; set; }
```
[the remark this answers is not in the document]
Comment (boma96) [reply src/ConductorSharp.Engine/Model/TerminateTaskModel.cs:19]: terminationStatus input parameter must be constant (i.e. either COMPLETED or FAILED). Specifying this parameter as an output of some task is not allowed (Registration of workflow will fail).

[the remark this answers is not in the document]
Comment (boma96) [reply src/ConductorSharp.Engine/Model/TerminateTaskModel.cs:19]: As for enum values, currently, they would be converted to numeric values. Newtonsoft serializer can be configured to serialize enum as a string but there is this line that prevents me from doing that.

```
  // TODO: Make this not required, as we want to allow the user to init client as he likes
  _restClient.UseNewtonsoftJson();
```

UseNetwonsoftJson method has an argument that allows us to configure serialization. Are we gonna remove this line as TODO comment says?

An alternative would be to detect these special enum constants in expression code and convert them manually.

[the remark this answers is not in the document]
Comment (boma96) [reply src/ConductorSharp.Engine/Model/TerminateTaskModel.cs]: @ognjenkatic Replaced static class with an enum, added code which handles enum constants. Also added a way for the library user to specify how enums are mapped.

**Review by ognjenkatic — APPROVED** [review]
LGTM

**Pull request merged**